dispatch/static/manager/src/js/pages/Articles/ArticleIndexPage.js

```diff
@@ -1,190 +1,72 @@
 import React from 'react'
 import { connect } from 'react-redux'
-import DocumentTitle from 'react-document-title'
-
-import { Link } from 'react-router'
-
-import { Intent } from '@blueprintjs/core'
 
+import ItemIndexPage from '../ItemIndexPage'
 import articlesActions from '../../actions/ArticlesActions'
 import { humanizeDatetime } from '../../util/helpers'
 
-import { LinkButton } from '../../components/inputs'
-import ItemList from '../../components/ItemList'
-
-const DEFAULT_LIMIT = 15
-
-class ArticlesPageComponent extends React.Component {
-
-  constructor(props) {
-    super(props)
-
-    this.handleDeleteArticles = this.handleDeleteArticles.bind(this)
-    this.handleSearchArticles = this.handleSearchArticles.bind(this)
-  }
-
-  getQuery() {
-    var query = {
-      limit: DEFAULT_LIMIT,
-      drafts: true,
-      offset: (this.getCurrentPage() - 1) * DEFAULT_LIMIT
-    }
-
-    // If section is present, add to query
-    if (this.props.location.query.section) {
-      query.section = this.props.location.query.section
-    }
-
-    // If search query is present, add to query
-    if (this.props.location.query.q) {
-      query.q = this.props.location.query.q
-    }
-
-    return query
-  }
-
-  getCurrentPage() {
-    return parseInt(this.props.location.query.page, 10) || 1
-  }
-
-  getTotalPages() {
-    return Math.ceil(
-      parseInt(this.props.articles.count, 10) / DEFAULT_LIMIT
-    )
-  }
-
-  componentWillMount() {
-    // Fetch articles
-    this.props.clearArticles()
-    this.props.clearSelectedArticles()
-    this.props.listArticles(this.props.token, this.getQuery())
-  }
-
-  componentDidUpdate(prevProps) {
-
-    if (this.isNewSection(prevProps, this.props) || this.isNewQuery(prevProps, this.props)) {
```

Comment:
I *really* like the abstraction provided by `ItemIndexPage`, but in this case we lose the `isNewSection` check, which prevents the user from switching between different section tabs. Can we maybe add an optional prop to `ItemIndexPage` that allows us to extend this function?